### DISCUSSION TITLE
Hello, Why remove Plugins functionality?

### DISCUSSION THREAD

**Discussion by dppo** [jpudysz/react-native-unistyles#263] [Q&A]
How can I intercept and modify styles globally without Plugins functionality?

**Comment by jpudysz** [comment]
New architecture for Unistyles moves parsing logic to C++. There won't be any parsing on JS side. I was able to support all other features.

In order to make plugins works I would need to parse styles one more time (in JS). From obvious reason it's not performant. 

Whatever is possible with plugins can be done with theme or using functions directly in the styles.

What's your use case ?

**Reply by dppo** [reply]
One of my requirements is to use fixed px layout on regular-sized phones (like iPhone 8), but apply global scaling, including size and font, on smaller screens (such as iPhone 5 or foldable screens). Previously, it was easy to achieve this through a plugin, allowing me to maintain excellent display effects on small screens without changing the existing code.

**Reply by jpudysz** [reply]
Can you show me your plugin code?

**Reply by dppo** [reply]
This is my feature code.
```
const screenAdapterPlugin: UnistylesPlugin = {
  name: 'screenAdapterPlugin',
  onParsedStyle: (key, styles, runtime) => {
    if (key.endsWith('_unscaled')) return styles;

    const adapterScale = screenAdapterScale({
      width: runtime.screen.width,
      designWidth: isDesktop(runtime.screen.width, runtime.screen.height) ? 820 : 375,
      maxScreenAdapterScale: 1,
      minScreenAdapterScale: 0.8,
    });

    const pairs = Object.entries(styles).map(([key, value]) => {
      const isNumber = typeof value === 'number';
      const ignoredStyles = ['flex', 'shadowOpacity', 'maxWidth', 'maxHeight', 'minWidth', 'minHeight'];
      if (!isNumber || ignoredStyles.includes(key)) return [key, value];
      return [key, value * adapterScale];
    });

    return Object.fromEntries(pairs);
  },
};

export const screenAdapterScale: AppUtils['screenAdapterScale'] = args => {
  return Math.max(args.minScreenAdapterScale, Math.min(args.width / args.designWidth, args.maxScreenAdapterScale));
};
```

**Reply by jpudysz** [reply]
I will think about it, thansk

**Comment by 321paranoiawhy** [comment]
Hey, bro, since 2.9.0, plugin feature has been deprecated, so something shoule be updated in [doc](https://www.unistyl.es/reference/plugins/) at the begging such as: do not use this feature in version above 2.9.0. Hope the updates, thanks in advance.

**Reply by jpudysz** [reply]
But you're notified as soon as you try to use it 😉

**Comment by yvbeek** [comment]
@jpudysz First of all, thanks for all of your hard work on this library!

With the plugin system now deprecated, it would be great if you could point out in the docs best practices on how to handle some common scenarios without plugins.

In my app I would like to override the default font that the app is using. Plugins seem like a perfect way to do that. It would be very error prone if I need to change that to theme function calls or have to rely on the other devs to supply the correct font styles everywhere.

Perhaps for performance we could have a plugin system on the native side? I wouldn't mind adding a bit of native code to handle this if that makes it way way more performant.
